client/components/theme-type-badge/tooltip.tsx

```diff
@@ -106,11 +107,17 @@ const ThemeTypeBadgeTooltip = ( {
 		} );
 	}, [ themeId ] );
 
+	const premiumPlan = plans?.data?.[ PLAN_PREMIUM ];
 	let message;
 	if ( isLockedStyleVariation ) {
-		message = translate(
-			'Unlock this style, and tons of other features, by upgrading to a %(premiumPlanName)s plan.',
-			{ args: { premiumPlanName: plans?.data?.[ PLAN_PREMIUM ]?.productNameShort ?? '' } }
+		message = createInterpolateElement(
+			// Translators: %(premiumPlanName)s is the name of the premium plan that includes this theme. Examples: "Explorer" or "Premium".
+			translate(
+				'Unlock this style, and tons of other features, by upgrading to a <Link>%(premiumPlanName)s plan</Link>.',
+				{ args: { premiumPlanName: premiumPlan?.productNameShort || '' }, textOnly: true }
+			),
+			// @ts-expect-error -- Property children is missing, but provided by the createInterpolateElement function.
+			{ Link: <ThemeTierBadgeCheckoutLink plan={ premiumPlan?.planSlug } /> }
```

Comment:
I think in this case I would use ThemeTypeBadgeTooltipUpgradeLink which is the non-theme-tier version of ThemeTierBadgeCheckoutLink. They should both work the same, it just seems...safer not to mix-and-match.

Reply:
Thanks for pointing it out, addressed on https://github.com/Automattic/wp-calypso/pull/85955/commits/5c002eb96749c03c2aadd7e0aada89cc7dd85da3.